### PR TITLE
VAR-419 | Implement DateRangePicker component

### PR DESCRIPTION
**Pull request by ahie**
## Description :sparkles:

Adds a component for selection of date ranges to reserve.

### Closes :no_good_woman:

[VAR-419](https://helsinkisolutionoffice.atlassian.net/browse/VAR-419)

### Automated tests :gear:️

- `src/common/date-range/__tests__/DateRangePicker.test.js`
- `src/common/date-range/__tests__/utils.test.js`
- `src/common/date-range/legend/__tests__/DateRangeLegend.test.js`
- `src/common/date-range/summary/__tests__/DateRangeSummary.test.js`

### Manual testing :construction_worker_man:

As the project does not use storybook, you will need to manually add the component somewhere in the page. Here's an example I used while developing the component:

```
          <DateRangePicker
            endingTime={{ hours: 11, min: 0 }}
            fullDay={false}
            maxDays={7}
            minDays={3}
            openingPeriods={[
              { from: new Date(2020, 5), to: new Date(2020, 8, 17) },
              { from: new Date(2020, 8, 20), to: new Date(2021, 1) },
            ]}
            renderChildComponent={selectedDateRange => (
              <>
                <DateRangeSummary
                  endDate={selectedDateRange.to}
                  startDate={selectedDateRange.from}
                />
                <DateRangeLegend />
              </>
            )}
            reservations={[
              {
                from: new Date(2020, 8, 26),
                to: new Date(2020, 8, 28),
              },
            ]}
            startingDays={[
              new Date(2020, 8, 10),
              new Date(2020, 8, 20),
              new Date(2020, 8, 26),
            ]}
            startingTime={{ hours: 15, min: 0 }}
          />
```

## Screenshots :camera_flash:

Screenshot 2020-08-14 at 10 26 39

Screenshot 2020-08-14 at 10 27 56

## Additional notes :spiral_notepad:

TODO: Need to fix Swedish translations.

### CODE REVIEW

File: src/common/date-range/legend/DateRangeLegend.js [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import PropTypes from 'prop-types';
+
+import injectT from '../../../../app/i18n/injectT';
+
+const DateRangeLegend = (props) => {
+  const { t } = props;
+  return (
+    <div className="app-DateRangeLegend">
+      <div className="legend-item">
+        <div className="color-box color-box__available">01</div>
```
Comment (Nizar-Rahme) [inline src/common/date-range/legend/DateRangeLegend.js:11]: Prefixing these classNames with `app-DateRangeLegend` would reduce the possibility of conflicting with other classNames.

File: src/common/date-range/DateRangePicker.js [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+      return;
+    }
+
+    const range = DateUtils.addDayToRange(day, selectedDateRange);
+    range.from && range.from.setHours(startingTime.hours, startingTime.min);
+    range.to && range.to.setHours(endingTime.hours, endingTime.min);
+    setSelectedDateRange(range);
+
+    if (isValidRange(range)) {
+      const validationErrors = validateRange(range, {
```
Comment (Nizar-Rahme) [inline src/common/date-range/DateRangePicker.js:68]: The function name `isValidRange` confused me a little bit especially that we called `validateRange` after it.
Would `isRange` be more descriptive?

Comment (ahie) [reply src/common/date-range/DateRangePicker.js]: Changed to `isRange`.

File: src/common/date-range/legend/DateRangeLegend.js (hunk 1 above)
Comment (ahie) [reply src/common/date-range/legend/DateRangeLegend.js]: Done :+1:

**Pull request merged**